pyproject.toml

```
@@ -10,58 +14,30 @@ documentation = "https://haptools.readthedocs.io"
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = ">=3.7,<3.11"
+python = ">=3.7,<4.0"
 numpy = ">=1.20.0"
 click = ">=8.0.3"
 pysam = ">=0.19.0"
 cyvcf2 = ">=0.30.14"
 matplotlib = ">=3.5.1"
 Pgenlib = ">=0.90.1"
 
-# docs
-# these belong in dev-dependencies, but RTD doesn't support that yet -- see
-# https://github.com/readthedocs/readthedocs.org/issues/4912
-Sphinx = { version = ">=4.3.2", optional = true }
-sphinx-autodoc-typehints = { version = ">=1.12.0", optional = true }
-sphinx-rtd-theme = { version = ">=1.0.0", optional = true }
-numpydoc = { version = ">=1.1.0", optional = true }
-sphinx-click = { version = ">=3.0.2", optional = true }
-
-[tool.poetry.dev-dependencies]
-
-# docs
-# these belong in dev-dependencies, but RTD doesn't support that yet -- see
-# https://github.com/readthedocs/readthedocs.org/issues/4912
-Sphinx = { version = ">=4.3.2" }
-sphinx-autodoc-typehints = { version = ">=1.12.0" }
-sphinx-rtd-theme = { version = ">=1.0.0" }
-numpydoc = { version = ">=1.1.0" }
-sphinx-click = { version = ">=3.0.2" }
+[tool.poetry.group.docs.dependencies]
+Sphinx = ">=4.3.2"
+sphinx-autodoc-typehints = ">=1.12.0"
+sphinx-rtd-theme = ">=1.0.0"
+numpydoc = ">=1.1.0"
+sphinx-click = ">=3.0.2"
 
+[tool.poetry.group.tests.dependencies]
 pytest = ">=6.2.5"
 black = ">=22.3.0"
 ipython = ">=7.30.1"
-nox = ">=2022.8.7"
-nox-poetry = ">=1.0.1"
+nox = ">=2022.11.21"
+nox-poetry = ">=1.0.2"
```

```suggestion
nox-poetry = ">=1.0.3"
```

this should probably match whatever we have in the constraints file

note that the lock file will also need to be updated after this